echopop/extensions/diagnostics.py

```diff
@@ -0,0 +1,510 @@
+                        }
+                    )
+                }
+            )
+
+        # Initialize `panel` extension
+        pn.extension()
+
+        # Initialize widgets
+        # ---- Variable options
+        var_options = ["Biomass", "Biomass density"]
+        # ---- Get column names
+        var_columns = {"Biomass": "total", "Biomass density": "density"}
+        # ---- y-axis names
+        y_axis_label = {
+            "Biomass": "Biomass (kmt) [mean ± 95% CI]",
+            "Biomass density": "Biomass density (kmt nmi⁻²) [mean ± 95% CI]",
+        }
+        # ---- Create dropdown selector
+        var_dropdown = pn.widgets.Select(name="Variable", options=var_options)
+
+        # Create plotting function
+        @pn.depends(selection=var_dropdown)
+        def update_plot(selection):
+
+            # Parse dataset
+            data = DATAINPUT[selection]
+            # ---- Get data column
+            value = var_columns[selection]
+
+            # Initialize plotting layers
+            plot_layers = []
+
+            # Plot the data
+            # ---- Kriging
+            if "kriging" in data:
+                # ---- Add errorbar
+                kriging_pt_err = hv.Segments(
+                    data["kriging"], kdims=["stratum", "lower", "stratum", "upper"], label="Kriging"
+                ).opts(color="red")
+                # ---- Point estimates
+                kriging_pts = hv.Points(
+                    data["kriging"], kdims=["stratum", value], label="Kriging"
+                ).opts(
+                    size=12,
+                    color="red",
+                    ylim=(0.0, np.inf),
+                    xticks=data["kriging"]["stratum"].tolist(),
+                )
+                # ---- Add to layer
+                plot_layers.append(kriging_pt_err * kriging_pts)
+                # ---- Transect
+                if "transect" in data:
+                    # ---- Add errorbar
+                    transect_pt_err = hv.Segments(
+                        data["transect"],
+                        kdims=["stratum", "lower", "stratum", "upper"],
+                        label="Transect",
+                    ).opts(color="black")
+                    # ---- Point estimates
+                    transect_pts = hv.Points(
+                        data["transect"],
+                        kdims=["stratum", var_columns[selection]],
+                        label="Transect",
+                    ).opts(
+                        size=12,
+                        color="black",
+                    )
+                    # ---- Add to layer
+                    plot_layers.append(transect_pt_err * transect_pts)
+                # ---- Layer the data
+                return hv.Overlay(plot_layers).opts(
+                    height=600, width=800, xlabel="INPFC stratum", ylabel=y_axis_label[selection]
+                )
```

Comment:
These are indented -- are the indents intentional? If so, then transect data will only be plotted if kriging analysis has been run - this would contradicts with the multiple `if "kriging" in self.data.results:` and `if "transect" in self.data.results:` checks you did earlier ([see my comments here](https://github.com/OSOceanAcoustics/echopop/pull/304/files?diff=unified&w=0#r1894709099)).